packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx

```diff
@@ -149,6 +152,26 @@ export class TextFieldBase extends BaseComponent<ITextFieldProps, ITextFieldStat
     if (newProps.defaultValue !== this.props.defaultValue && newProps.value === undefined) {
       this._setValue(newProps.defaultValue);
     }
+
+    // Text field is changing between single- and multi-line. After the change is complete,
+    // we'll need to reset focus and selection/cursor.
+    if (!!newProps.multiline !== !!this.props.multiline && this.state.isFocused) {
+      this._resetFocusAfterInputTypeChange = true;
+      this._selectionBeforeInputTypeChange = [this.selectionStart, this.selectionEnd];
+    }
+  }
+
+  public componentDidUpdate(): void {
+    if (this._resetFocusAfterInputTypeChange) {
+      // The text field has just changed between single- and multi-line, so we need to reset focus
+      // and selection/cursor.
+      this._resetFocusAfterInputTypeChange = false;
+      this.focus();
```

Comment:
I vaguely remember an issue in some previous code I worked on where refs weren't defined right after render, so you had to use setTimeout to delay before (for example) using a ref to set focus. Is anyone familiar with that issue? (it's possible the issue only applied in componentDidMount, not componentDidUpdate)

Reply:
Refs are [set after mounting and are "guaranteed to be up-to-date before componentDidMount or componentDidUpdate fires"](https://reactjs.org/docs/refs-and-the-dom.html#callback-refs) so this should be ok.

Reply:
Oh great, thanks!